modules/core/shared/src/test/scala/ciris/ConfigFieldSpec.scala

```diff
@@ -0,0 +1,132 @@
+package ciris
+
+import cats.effect.IO
+import cats.syntax.all._
+import munit.CatsEffectSuite
+import munit.ScalaCheckEffectSuite
+
+final class ConfigFieldSpec extends CatsEffectSuite with ScalaCheckEffectSuite with Generators {
+  def check[F[_], A](config: ConfigValue[F, A], expected: List[ConfigField]): Unit =
+    assertEquals(config.fields, expected)
+
+  test("ConfigValue.pure.fields") {
+    check(ConfigValue.pure(ConfigEntry.loaded(None, "test")), Nil)
+  }
+
+  test("ConfigValue.environment.fields") {
+    check(env("ENV_VARIABLE"), List(ConfigField.required(ConfigKey.env("ENV_VARIABLE"))))
+  }
+
+  test("ConfigValue.prop.fields") {
+    check(prop("user.dir"), List(ConfigField.required(ConfigKey.prop("user.dir"))))
+  }
+
+  test("ConfigValue.prop.default.fields") {
+    check(
+      prop("user.dir").default("~"),
+      List(ConfigField.optional(ConfigKey.prop("user.dir"), Some("~")))
+    )
+  }
+
+  test("ConfigValue.prop.default.default.fields") {
+    check(
+      prop("user.dir").default("/").default("~"),
+      List(ConfigField.optional(ConfigKey.prop("user.dir"), Some("~")))
+    )
+  }
+
+  test("ConfigValue.option.fields") {
+    check(
+      prop("user.dir").option,
+      List(ConfigField.optional(ConfigKey.prop("user.dir"), None))
+    )
+  }
+
+  test("ConfigValue.option.option.fields") {
+    check(
+      prop("user.dir").option.option,
+      List(ConfigField.optional(ConfigKey.prop("user.dir"), None))
+    )
+  }
+
+  test("ConfigValue.option.default.fields") {
+    check(
+      prop("user.dir").option.default(Some("discarded since option always returns a value")),
```

Comment:
I would expect this to work similar to how default values currently work, where later defaults override earlier defaults. The `option` is treated as `.map(_.some).default(None)`, so this is `.map(_.some).default(None).default(Some(...))` and the default value is `Some(...)`. Maybe I'm missing something here?

Reply:
> `.map(_.some).default(None).default(Some(...))`

So this code is supposed to return `Some(...)` if there is no value passed IIUC?

Reply:
Yes, exactly.

Reply:
Ok. So I indeed implemented the wrong behaviour. Currently, `configA.default(b).default(x)` will only return `configA`'s value or `b`.